doc/manual/chapters/translation.tex

```diff
@@ -220,27 +568,47 @@ \subsubsection{Maps}
 \noindent
 \begin{minipage}[t]{\textwidth}
 \begin{lstlisting}[language=golang]
-import "pgo/datatypes"
-
-var S datatypes.Set = datatypes.NewSet(1, 2)
-var f datatypes.Map = datatypes.MapsTo(func(x int, y int) int {
-		return x + y
-	}, S, S)
-var a int = f.Get(datatypes.NewTuple(2, 2))
+S := []int{1, 2}
+function := make([]struct {
+	key struct {
+		e0 int
+		e1 int
+	}
+	value int
+}, 0, len(S)*len(S))
+for _, x := range S {
+	for _, y := range S {
+		function = append(function, struct {
+			key struct {
+				e0 int
+				e1 int
+			}
+			value int
+		}{key: struct {
+			e0 int
+			e1 int
+		}{x, y}, value: x + y})
+	}
+}
+f := function
+key := struct {
+	e0 int
+	e1 int
+}{2, 2}
+index := sort.Search(len(f), func(i int) bool {
+	return !(f[i].key.e0 < key.e0 || f[i].key.e0 == key.e0 && f[i].key.e1 < key.e1)
+})
+a := f[index].value
 \end{lstlisting}
 \captionof{figure}{Compiled Go}
 \end{minipage}
 
-(Behind the scenes, the \texttt{datatypes.Map} uses tuples as keys.)
-
 \subsubsection{Tuples}
-PlusCal tuples are used in several different contexts, so variables involving tuples must be annotated with their types. Tuples can store homogeneous data, in which case they correspond to Go slices. Tuple components may be of different types, which correspond to \texttt{datatypes.Tuple}s. Most often in multiprocess algorithms, tuples are used for communication, in which case \texttt{datatypes.Tuple}s are used. PlusCal tuples are 1-indexed, but Go tuples and slices are 0-indexed, so 1 is subtracted from all indices in Go.
+PlusCal tuples are used in several different contexts, so variables involving tuples may have different inferred types depending on their use. Tuples can store homogeneous data, in which case they correspond to Go slices. Tuple components may be of different types, which correspond to Go structs. PlusCal tuples are 1-indexed, but Go tuples and slices are 0-indexed, so 1 is subtracted from all indices in Go.
```

Comment:
One question that raises from this description is what happens when you use the `Append` operator for heterogenous tuples.